docs/data-masking-function-list.md

```diff
@@ -0,0 +1,888 @@
+    | 520754XXXXXX4848                         |
+    +------------------------------------------+
+    ```
+
+## mask_ssn(str [,mask_char])
+
+
+
+Returns a masked United States Social Security Number(SSN). The mask replaces the SSN number with the specified character except for the last four digits.
+
+### Parameters
+
+| Parameter | Optional | Description | Type |
+| --- | --- | --- | --- |
+| str | No | The string to be masked | String |
+| mask_char | Yes | The masking character | String |
+
+The `str` accepts either of the following:
+
+* Nine integers, no separator symbol
+* Nine integers in the `***-**-****` pattern. The `-` is a separator character.
+
+If the `mask_char` is not specified, the default is `*`.
+
+### Returns
+
+A string with the selected characters masked by a specified `mask_char` or that parameter's default value. Returns a NULL value if the `str` is in an incorrect length. A multi-byte character `str` is not masked.
+
+### Example
+
+```{.bash data-prompt="mysql>"}
+mysql> SELECT mask_ssn('555-55-5555');
+```
+
+??? example "Expected output"
+
+    ```{.text .no-copy}
+    +-------------------------+
+    | mask_ssn('555-55-5555') |
+    +-------------------------+
+    | ***-**-5555             |
+    +-------------------------+
+    ```
+
+## mask_uk_nin(str [,mask_char])
+
+
+
+Returns a masked a United Kingdom National Insurance Number (NIN). The mask replaces the NIN number with the specified character except for the first two digits.
+
+### Parameters
+
+| Parameter | Optional | Description | Type |
+| --- | --- | --- | --- |
+| str | No | The string to be masked | String |
+| mask_char | Yes | The masking character | String |
+
+The `str` accepts an alpha-numeric string and does not check format.
+
+The `str` can use any separator character.
+
+If the `mask_char` is not specified, the default is `*`.
+
+### Returns
+
+Returns a string with the selected characters masked by a specified `mask_char` or that parameter's default value. 
+
+An error occurs if the `str` parameter is not the correct length. 
+
+Returns a NULL value if the `str` is in an incorrect format. A multi-byte character `str` is not masked.
+
+### Example
+
+```{.bash data-prompt="mysql>"}
+mysql> SELECT mask_uk_nin ('CT 26 46 83 D');
+```
+
+??? example "Expected output"
+
+    ```{.text .no-copy}
+    +------------------------------------+
+    | mask_uk_nin('CT 26 46 83 D')       |
+    +------------------------------------+
+    | CT ** ** ** *                      |
+    +------------------------------------+
+    ```
+
+## mask_uuid(str [,mask_char])
+
+
+
+Masks a Universally Unique Identifier (UUID).
+
+### Parameters
+
+| Parameter | Optional | Description | Type |
+| --- | --- | --- | --- |
+| str | No | The string to be masked | String |
+| mask_char | Yes | The masking character | String |
+
+The `str` format is `********-****-****-****-************`.
+
+If the `mask_char` is not specified, the default is '*'.
+
```

Comment:
mask_char can be a multibyte character in any character set (not necessarily identical to the character set of str)